crates/worktree/src/worktree.rs

```diff
@@ -3224,11 +3224,74 @@ async fn is_git_dir(path: &Path, fs: &dyn Fs) -> bool {
 async fn build_gitignore(abs_path: &Path, fs: &dyn Fs) -> Result<Gitignore> {
     let contents = fs.load(abs_path).await?;
     let parent = abs_path.parent().unwrap_or_else(|| Path::new("/"));
+
     let mut builder = GitignoreBuilder::new(parent);
     for line in contents.lines() {
         builder.add_line(Some(abs_path.into()), line)?;
     }
-    Ok(builder.build()?)
+
+    builder.build()?;
+
+    let mut combined_builder = GitignoreBuilder::new(parent);
+
+    let home = home_dir();
+    let repo_root = abs_path.parent().unwrap_or_else(|| Path::new("/"));
+    let repo_git_path = repo_root.join(".git");
+    let repo_git_config_path = repo_git_path.join("config");
+    
+    let mut global_ignore_path = None;
+    if let Ok(config_output) = std::process::Command::new("git")
+        .args(["config", "--file", &repo_git_config_path.to_string_lossy(), "core.excludesFile"])
+        .output()
+    {
+        if config_output.status.success() {
+            let path_str = String::from_utf8_lossy(&config_output.stdout).trim().to_string();
+            if !path_str.is_empty() {
+                let path = PathBuf::from(path_str);
+                global_ignore_path = Some(path);
+            }
+        }
+    }
+    
+    if global_ignore_path.is_none() {
+        if let Ok(config_output) = std::process::Command::new("git")
+            .args(["config", "--global", "core.excludesFile"])
+            .output()
+        {
+            if config_output.status.success() {
+                let path_str = String::from_utf8_lossy(&config_output.stdout).trim().to_string();
+                if !path_str.is_empty() {
+                    let path = PathBuf::from(path_str);
+                    global_ignore_path = Some(path);
+                }
+            }
+        }
+    }
```

Comment:
Instead of calling `git config` twice with different flags, I think we should just do `git config core.excludesFile`, which will merge the global and local configs in the right way for us.